### PR TITLE
Fix Redundant Overrides

### LINKED ISSUE #129: Remove redundant type compiler overrides
The DB2TypeCompiler class needlessly overrides various types with the same definition from the GenericTypeCompiler parent class.

eg.
```python
    def visit_SMALLINT(self, type_, **kw):
        return "SMALLINT"

    def visit_INT(self, type_):
        return "INT"

    def visit_BIGINT(self, type_, **kw):
        return "BIGINT"
```

These are all exactly the same as in the [parent class](https://github.com/sqlalchemy/sqlalchemy/blob/fb81f9c8d914f9911925dd3f4e77d7fc374b267c/lib/sqlalchemy/sql/compiler.py#L4502-L4509) so why are we overriding them? It just adds noise to the code.

### PR DESCRIPTION
**Pull request by markdirish**
Remove overrides for functions that are identical in the base class.

Change the `INT` functions to `INTEGER` functions (they are aliases in both Db2 and sqlalchemy) to use base class implementations.

Fixes #129

### CODE REVIEW

File: sqlalchemy_ibmi/base.py [hunk 1]
```diff
@@ -258,40 +258,16 @@ class XML(sa_types.Text):
 class DB2TypeCompiler(compiler.GenericTypeCompiler):
     """IBM i Db2 Type Compiler"""
 
-    def visit_TIMESTAMP(self, type_, **kw):
-        return "TIMESTAMP"
-
-    def visit_DATE(self, type_, **kw):
-        return "DATE"
-
-    def visit_TIME(self, type_, **kw):
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py:267]: Does the upstream TIME type support precision? If so, we should keep this and raise an exception if the user specifies a precision.

Comment (markdirish) [reply sqlalchemy_ibmi/base.py:267]: It looks like it does not

File: sqlalchemy_ibmi/base.py [hunk 2]
```diff
@@ -258,40 +258,16 @@ class XML(sa_types.Text):
 class DB2TypeCompiler(compiler.GenericTypeCompiler):
     """IBM i Db2 Type Compiler"""
 
-    def visit_TIMESTAMP(self, type_, **kw):
-        return "TIMESTAMP"
```
[the remark this answers is not in the document]
Comment (markdirish) [reply sqlalchemy_ibmi/base.py:262]: It looks like it does not

File: sqlalchemy_ibmi/base.py [hunk 3]
```diff
@@ -335,36 +311,15 @@ def visit_numeric(self, type_, **kw):
     def visit_datetime(self, type_, **kw):
         return self.visit_TIMESTAMP(type_)
 
-    def visit_date(self, type_, **kw):
```
[the remark this answers is not in the document]
Comment (markdirish) [reply sqlalchemy_ibmi/base.py:338]: I think that the lowercase names are the `visit` calls for the SQLAlchemy classes. Classes like `DateTime` and `Date`, which are used internally with the SQLAchemy code. There is a lot of internal logic in there for doing SQLAlchemy "things". The uppercase classes are the SQL Types, like `DATETIME`, `DATE`, etc., which appear to be dbapi/DBMS specific, and are really bare, probably just to generate type names for queries.

I found this looking through `lib/sqlachemy/sql/sqltypes.py`

**Comment by kadler** [comment]
Sorry, I forgot this PR existed and ended up doing it myself in 92e1df5affa1327b60d777d73bc6d97c3cf6f0fd.

**Pull request closed without merging**